eng/scripts/New-BuildInfo.ps1

```diff
@@ -0,0 +1,428 @@
+#!/bin/env pwsh
+#Requires -Version 7
+
+[CmdletBinding()]
+param(
+    [parameter(Mandatory)]
+    [ValidateSet('none', 'internal', 'public')]
+    [string] $PublishTarget,
+    [parameter(Mandatory)]
+    [int] $BuildId,
+    [string] $OutputPath,
+    [string] $ServerName,
+    [switch] $IncludeNative,
+    [switch] $TestPipeline
+)
+
+. "$PSScriptRoot/../common/scripts/common.ps1"
+. "$PSScriptRoot/helpers/PathHelpers.ps1"
+$RepoRoot = $RepoRoot.Path.Replace('\', '/')
+$isPipelineRun = $env:TF_BUILD -eq 'true'
```

Comment:
I would make this a `[switch]` parameter with the default being set by the `TF_BUILD` check, that way it's easier to test CI mode locally when development requires it.